Source/DafnyDriver/DafnyDriver.cs

```diff
@@ -729,4 +745,34 @@ private record TargetPaths(string Directory, string Filename) {
     #endregion
 
   }
+
+  class NoCompiler : Compiler {
```

Comment:
When a command doesn't use a compiler, there's still some pointless option processing code that depends on a compiler being defined. Previously the C# compiler was implicitly used in this case, but now I had to either use it explicitly or add this `NoCompiler`.

When I do another refactoring that replaces ICommandSpec with direct `System.CommandLine.Command` usage, I will also remove NoCompiler, since then the commands that don't use a compiler won't run the option processing code that currently uses the configured compiler.